src/SFA.DAS.Reservations.Application/AccountReservations/Services/AccountReservationService.cs

```diff
@@ -102,6 +110,58 @@ public async Task DeleteReservation(Guid reservationId)
             return reservations.Select(c=>c.Id).ToList();
         }
 
+        public async Task<Guid> ChangeOfParty(ChangeOfPartyServiceRequest request)
+        {
+            var existingReservation = await _reservationRepository.GetById(request.ReservationId);
+            if (existingReservation == null)
+            {
+                throw new EntityNotFoundException<Domain.Entities.Reservation>();
+            }
+
+            if (existingReservation.Status != (short)ReservationStatus.Confirmed &&
+                existingReservation.Status != (short)ReservationStatus.Change)

```

Comment:
The existing Reservation status can be **Change**, you should be able to go from employer a->b->c->n, each time calling change will look at the reservation passed. The only statuses that should be ignored are Completed and Deleted

Reply:
supported